### PR TITLE
Added support for DFS configuration using SPLUNK_ENABLE_DFS flag

### PR DESCRIPTION
**Pull request by mikedickey**
Note that this has various dependencies, including Spark being available
and installed in SPARK_HOME, Java being available and installed in
JAVA_HOME, a spark cluster available, and other things.

### CODE REVIEW

File: inventory/environ.py [hunk 1]
```diff
@@ -217,6 +217,11 @@ def overrideEnvironmentVars(vars_scope):
         vars_scope["splunk"]["multisite_replication_factor_total"] = os.environ.get('SPLUNK_MULTISITE_REPLICATION_FACTOR_TOTAL', vars_scope["splunk"].get("multisite_replication_factor_total", 1))
         vars_scope["splunk"]["multisite_search_factor_origin"] = os.environ.get('SPLUNK_MULTISITE_SEARCH_FACTOR_ORIGIN', vars_scope["splunk"].get("multisite_search_factor_origin", 1))
         vars_scope["splunk"]["multisite_search_factor_total"] = os.environ.get('SPLUNK_MULTISITE_SEARCH_FACTOR_TOTAL', vars_scope["splunk"].get("multisite_search_factor_total", 1))
+    #Used for DFS
+    vars_scope["splunk"]["dfs"]["enable"] = os.environ.get('SPLUNK_ENABLE_DFS', vars_scope["splunk"]["dfs"]["enable"])
+    vars_scope["splunk"]["dfs"]["dfw_num_slots_enabled"] = os.environ.get('SPLUNK_DFW_NUM_SLOTS_ENABLED', vars_scope["splunk"]["dfs"]["dfw_num_slots_enabled"])
+    vars_scope["splunk"]["dfs"]["spark_master_host"] = os.environ.get('SPARK_MASTER_HOST', vars_scope["splunk"]["dfs"]["spark_master_host"])
+    vars_scope["splunk"]["dfs"]["spark_master_webui_port"] = os.environ.get('SPARK_MASTER_WEBUI_PORT', vars_scope["splunk"]["dfs"]["spark_master_webui_port"])
```
Comment (nwang92) [inline inventory/environ.py]: Do we also want to expose setting `dfw_num_slots` or `dfc_num_slots` via environment vars?

Comment (lephino) [reply inventory/environ.py]: I think we should if we're allowing the var to turn it on

**Comment by bb03** [comment]
@mikedickey Can you retest this with the latest code and verify that the behavior is correct?

**Comment by mikedickey** [comment]
> @mikedickey Can you retest this with the latest code and verify that the behavior is correct?

Done -- tested everything end-to-end and also ran all the docker-splunk tests using develop. Everything looks good.

**Pull request merged**